### PR TITLE
Add build target for packaging tsserver as a library

### PR DESCRIPTION
**Pull request by weswigham**
Following the discussion on #3513. this is the first half of the proposed changes - only those required to package the service as a library, combined with feedback from the same PR.

### CODE REVIEW

File: src/server/editorServices.ts [hunk 1]
```diff
@@ -1360,10 +1360,11 @@ module ts.server {
             }
         }
 
-        static fromString(script: string) {
+        static fromString(host: System, script: string) {
             var svc = new ScriptVersionCache();
```
Comment (mhegazy) [inline src/server/editorServices.ts]: ServerHost

File: src/server/tsconfig.json [hunk 2]
```diff
@@ -12,6 +12,7 @@
         "editorServices.ts",
         "protocol.d.ts",
         "server.ts",
+        "nodeimpl.ts",
```
Comment (mhegazy) [inline src/server/tsconfig.json]: note needed.

**Comment by mhegazy** [comment]
thanks. looks good. 2 small comments and we are ready to go.

**Comment by weswigham** [comment]
Got it.

**Comment by mhegazy** [comment]
thanks!

**Pull request merged**